osm.py

```diff
@@ -531,11 +535,42 @@ def diff_settings(dest, src):
         # If neither exist, nothing to do, nothing to say. Move along.
 
 
+def backup_vault(vault_path):
+    print('#', vault_path)
+
+    vault_path = Path(vault_path)
+
+    backup_info = must_get_key(OSM_CONFIG, 'backup', 'in OSM configuration')
+    backup_dir_name = Path(must_get_key(backup_info, 'location', 'in "backup" section of OSM configuration'))
+    backup_formats = must_get_key(backup_info, 'archive_format_priority_list', 'in "backup" section of OSM configuration')
+
+    backup_from_dir = vault_path / '.obsidian'
+    backup_full_dir = Path(vault_path) / backup_dir_name
+    backup_full_dir.mkdir(exist_ok=True)
+    backup_file = Path(vault_path) / backup_dir_name / datestring()
+    for backup_format in backup_formats:
+        try:
+            backup_file = shutil.make_archive(backup_file, backup_format, root_dir=backup_from_dir)
+            if backup_file:
+                print("Backup made:", backup_file)
```

Comment:
If we like how this works, I'll update to have versbose and dry-run implemented. The trick with dry-run though is that until we try to call `make_archive` we don't know for sure sure sure if that format is actually supported.
There is https://docs.python.org/3/library/shutil.html#shutil.get_archive_formats but I am not sure if those are guaranteed to work, or if they are just potentially available...

Reply:
`shutil.make_archive()` takes a `dry_run` boolean. ;-)

Reply:
D'oh!  So it does! I think I knew that when I first found it, but then forgot, thanks!